### PR TITLE
Update bash-completion

### PR DESCRIPTION
**Pull request by nak3**
This patch inclues:
* Complete the name of resource with `oc label`, `oc stop`, `oc expose`, `oc export`, `oc patch`.
* Add json or yaml filename with `-f` and `--filename` flags with `oc volumes` and `oc process`

### CODE REVIEW

**Comment by smarterclayton** [comment]
This breaks `-f -` to read from STDIN

```
the path "[-]" does not exist
!!! Error in hack/test-cmd.sh:257
  'oc process -f -' exited with status 0 1
Call stack:
  1: hack/test-cmd.sh:257 main(...)
```

File: pkg/cmd/cli/cmd/volume.go [hunk 1]
```diff
@@ -143,9 +144,11 @@ func NewCmdVolume(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.C
 			kcmdutil.CheckErr(err)
 		},
 	}
+	usage := "Filename, directory, or URL to file to use to edit the resource"
+	kubectl.AddJsonFilenameFlag(cmd, &filenames, usage)
+
 	cmd.Flags().StringVarP(&opts.Selector, "selector", "l", "", "Selector (label query) to filter on")
 	cmd.Flags().BoolVar(&opts.All, "all", false, "select all resources in the namespace of the specified resource types")
-	cmd.Flags().VarP(&opts.Filenames, "filename", "f", "Filename, directory, or URL to file to use to edit the resource.")
```
Comment (liggitt) [inline pkg/cmd/cli/cmd/volume.go]: how is the data getting into `opts.Filenames` so the command functions?

File: pkg/cmd/cli/cmd/env.go [hunk 2]
```diff
@@ -73,12 +73,14 @@ func NewCmdEnv(fullName string, f *clientcmd.Factory, in io.Reader, out io.Write
...
 	}
+	usage := "Filename, directory, or URL to file to use to edit the resource"
+	kubectl.AddJsonFilenameFlag(cmd, &filenames, usage)
+
 	cmd.Flags().StringP("containers", "c", "*", "The names of containers in the selected pod templates to change - may use wildcards")
 	cmd.Flags().VarP(&env, "env", "e", "Specify key value pairs of environment variables to set into each container.")
 	cmd.Flags().Bool("list", false, "Display the environment and any changes in the standard format")
 	cmd.Flags().StringP("selector", "l", "", "Selector (label query) to filter on")
 	cmd.Flags().Bool("all", false, "select all resources in the namespace of the specified resource types")
-	cmd.Flags().VarP(&filenames, "filename", "f", "Filename, directory, or URL to file to use to edit the resource.")
```
Comment (liggitt) [inline pkg/cmd/cli/cmd/env.go]: you can accomplish the same filename autocompletion with 
```
cmd.MarkFlagFilename("filename", "yaml", "yml", "json")
```

**Comment by danmcp** [comment]
@fabianofranz Review please

**Comment by fabianofranz** [comment]
@nak3 Please rebase. Other commands that could also make use of resource name completion: `oc annotate`, `oc env`, `oc edit`, `oc volume`. Make sure to run `hack/update-generated-completions.sh` to update the completions. Apart that, looks good.

**Comment by nak3** [comment]
Thank you @fabianofranz . Updated

**Comment by fabianofranz** [comment]
[test]

**Comment by fabianofranz** [comment]
Thanks @nak3. @smarterclayton the issue you mentioned is fixed. LGTM, [merge]

**Comment by fabianofranz** [comment]
Unrelated failure, [merge].

**Pull request merged**